### ISSUE TITLE
support for namespace packages with __init__.py file

### ISSUE THREAD

**Issue by analog-cbarber** [mkdocstrings/griffe#58]
Namespace packages with `__init__.py` files are not supported.

After working around the cyclic-import / recursion bug (#57),
my doc failed to build with an error like 

```
ERROR    -  mkdocstrings: garpy.cavr.core.dataset could not be found
ERROR    -  Error reading page 'api/dataset.md':
ERROR    -  Could not collect 'garpy.cavr.core.dataset'
```

The `garpy` and `garpy.cavr` packages are namespace packages that have `__init__.py` files looking like:

```python
# pylint: disable=missing-module-docstring
__path__ = __import__("pkgutil").extend_path(__path__, __name__) # type: ignore
```
It looks like griffe is assuming that namespace packages do not have `__init__.py` files:

https://github.com/mkdocstrings/griffe/blob/9bfdee41b49d98ae9fe8d51e037116c561c815af/src/griffe/finder.py#L141-L145

When I hacked that code to consider my package to be a namespace package I was finally able to get the doc to build.

I am not sure the best way for griffe to check the `__init__.py` file to verify that it is a namespace package.

Also see https://packaging.python.org/en/latest/guides/packaging-namespace-packages/

**Comment by pawamoy** [comment]
Thanks again for the report :slightly_smiling_face: 

Indeed, I only added support for native namespace packages :thinking: 
I fear supporting explicit namespace packages will be challenging :thinking:
Just by looking at comments in your example, it shows we can't do a naive comparison on the module contents.
To properly detect such namespace packages we would probably need to get the AST and minify it before comparing it with our own known ASTs.

**Comment by pawamoy** [comment]
I've opened https://github.com/pypa/sample-namespace-packages/issues/20, if you have some answers please share :smile:

**Comment by analog-cbarber** [comment]
You could always just add an option to explicitly declare namespace packages that could be specified in the mkdocs.yml file:

**Comment by pawamoy** [comment]
Sure, but I'd much prefer being able to properly detect such namespace packages automatically :smile: 
I need to think about it more.

**Comment by pawamoy** [comment]
I agree. Implementing an option is probably not easier though. It would impact mkdocstrings, mkdocstrings-python, and griffe. That's a lot of work.

I still need to think about it :open_hands:

**Comment by analog-cbarber** [comment]
Fair enough. Maybe you could just walk up the tree from any watch paths until you find a `setup.py`, `setup.cfg`, or `pyproject.tml` file. That should be your local source root.

**Comment by pawamoy** [comment]
I went for a naive solution based on regular expressions, using the recommended statements from the linked packaging docs.

**Comment by analog-cbarber** [comment]
Thanks. I will give it a try when it makes it to a release.

**Comment by pawamoy** [comment]
Released in 0.17.0!
